src/app/vault/vault.component.ts

```diff
@@ -223,9 +223,18 @@ export class VaultComponent implements OnInit, OnDestroy {
         });
     }
 
-    viewCipher(cipher: CipherView) {
+    async viewCipher(cipher: CipherView) {
         if (this.action === 'view' && this.cipherId === cipher.id) {
             return;
+        } else if (document.getElementsByClassName("ng-dirty").length > 0 && (this.action == 'add' || this.action === 'edit')) {
```

Comment:
What if an input outside of the add/edit forms is dirty? Like the search box?

Reply:
@kspearrin good catch! I'm not too familiar with angular, is there a way to check the `AddEdit` components inputs/fields only?

Reply:
You could just limit your `document` query to a section of the DOM tree only applicable to the add/edit pages.

Reply:
@kspearrin  Changed it to `document.getElementById('details').getElementsByClassName('ng-dirty')`